compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPackToDma.cpp

```diff
@@ -0,0 +1,359 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "air/Dialect/AIR/AIRDialect.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtDialect.h"
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Utils/IndexingUtils.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-amdaie-pack-to-dma"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Applies packing to a given input
+LogicalResult packDmaInputs(RewriterBase &rewriter,
+                            IREE::LinalgExt::PackOp packOp,
+                            SmallVector<Value> &offsets,
```

Comment:
I started doing it like this but it was requiring too may conversions. I see in IREE there are several places where these are 
SmallVector<OpFoldResult> but in those cases the API gives them as OpFolderResult from the subview op, however here, we are getting these from an allocOp as int64_t, so needed conversions in almost all the functions, for example  in the packing/unpacking logic I need integers to do arithmetic on the shapes and strides so would have to convert to integer and then back after, which seemed unnecessary. Let me know what you think. I did add a lambda to easily convert the integers to value in the end which helps with readability as well.

Reply:
It would be better to use `OpFoldResult`. https://github.com/shark-infra/llvm-project/blob/3c25a9de91e9c76ffba82939b19eafe3e60d51f7/mlir/include/mlir/Dialect/Utils/StaticValueUtils.h#L90-L91 has the functions to convert to integers. Another advantage is that you can easily handle dynamic cases by using `affine_map` to represent the math and then use `makeComposedAffineApply`.... Anyway, its fine either way.